src/persist-client/src/batch.rs

```diff
@@ -422,25 +459,50 @@ impl<T: Timestamp + Codec64> BatchParts<T> {
                 .writing_parts
                 .pop_front()
                 .expect("pop failed when len was just > some usize");
-            let () = handle
+            let () = match handle
                 .instrument(debug_span!("batch::max_outstanding"))
                 .await
-                .expect("part upload task failed");
+            {
+                Ok(()) => (),
+                Err(err) if err.is_cancelled() => (),
+                Err(err) => panic!("part upload task failed: {}", err),
+            };
             self.finished_parts.push(key);
         }
     }
 
     #[instrument(level = "debug", name = "batch::finish_upload", skip_all, fields(shard = %self.shard_id))]
-    async fn finish(self) -> Vec<String> {
+    pub(crate) async fn finish(self) -> Vec<String> {
         let mut keys = self.finished_parts;
         for (key, handle) in self.writing_parts {
-            let () = handle.await.expect("part upload task failed");
+            let () = match handle.await {
+                Ok(()) => (),
+                Err(err) if err.is_cancelled() => (),
+                Err(err) => panic!("part upload task failed: {}", err),
+            };
             keys.push(key);
         }
         keys
     }
 }
 
+pub(crate) fn truncate_batch<T: Timestamp>(
```

Comment:
nit: could be called `verify_truncate_batch()` or `check_truncate_batch()` or `is_truncate_batch()`, because the current name suggests to me that it does the truncation.

Reply:
👌